sound/soc/sof/sof-audio.h

```diff
@@ -205,6 +206,38 @@ const struct sof_ipc_pipe_new *snd_sof_pipeline_find(struct snd_sof_dev *sdev,
 void snd_sof_pcm_period_elapsed(struct snd_pcm_substream *substream);
 void snd_sof_pcm_period_elapsed_work(struct work_struct *work);
 
+void snd_sof_compr_fragment_elapsed(struct snd_compr_stream *cstream);
+void snd_sof_compr_fragment_elapsed_work(struct work_struct *work);
+
+static void inline snd_sof_period_elapsed(struct snd_sof_pcm_stream *sps)
+{
+	if (sps->substream && !sps->substream->runtime->no_period_wakeup) {
+		snd_sof_pcm_period_elapsed(sps->substream);
+		return;
+	}
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_COMPRESS)
+	if (sps->cstream)
+		snd_sof_compr_fragment_elapsed(sps->cstream);
+#endif
+}
+
+static void inline snd_sof_period_elapsed_work(struct work_struct *work)
+{
+	struct snd_sof_pcm_stream *sps =
+		container_of(work, struct snd_sof_pcm_stream, period_elapsed_work);
+
+	if (sps->substream) {
+		snd_sof_pcm_period_elapsed_work(work);
+		return;
+	}
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_COMPRESS)
+	if (sps->cstream)
+		snd_sof_compr_fragment_elapsed_work(work);
+#endif
```

Comment:
do we actually need to have a compilation flag in the core? Wondering if this shouldn't be a platform property?

Reply:
+1, maybe the cstream field isn't optional but it's always NULL if COMPRESS is disabled, and the function exists but is empty when COMPRESS is disabled.

Reply:
Yes, it makes sense. I will remove this.